plugins/modules/keycloak_role.py

```diff
@@ -269,10 +312,15 @@ def main():
         new_param_value = module.params.get(param)
         old_value = before_role[param] if param in before_role else None
         if new_param_value != old_value:
-            changeset[camel(param)] = new_param_value
+            if isinstance(param, dict):
+                changeset[camel(param)] = copy.deepcopy(new_param_value)
+            elif isinstance(param, list):
+                changeset[camel(param)] = new_param_value.copy()
+            else:
+                changeset[camel(param)] = new_param_value
```

Comment:
Why not simply:
```suggestion
            changeset[camel(param)] = copy.deepcopy(new_param_value)
```
?